hs_core/views/utils.py

```diff
@@ -21,9 +23,12 @@ def upload_from_irods(username, password, host, port, zone, irods_fname, res_fil
     try:
         irods_storage = IrodsStorage()
         irods_storage.set_user_session(username=username, password=password, host=host, port=port, zone=zone)
+        size = irods_storage.size(irods_fname)
+        if size > file_size_limit:
```

Comment:
I did refactor the code to remove duplication and added the common raise_file_size_exception() function which is used by both check_resource_files() and this irods upload function. Since we get file size differently between these two functions, I think this is good enough without suffering from the risk of code duplication. Let me know if you have further specific suggestions.